lib/carma/models/submission.rb

```diff
@@ -0,0 +1,60 @@
+# frozen_string_literal: true
+
+module CARMA
+  module Models
+    class Submission < Base
+      attr_reader :metadata
+      attr_accessor :data
+      attr_accessor :carma_case_id, :submitted_at
+
+      request_payload_key :data, :metadata
+
+      # Returns a new CARMA::Models::Submission built from a CaregiversAssistanceClaim.
+      #
+      #
+      # @param claim [CaregiversAssistanceClaim] A validated CaregiversAssistanceClaim
+      # @param metadata [Hash] Additional data that can be submitted along with the claim's form data
+      # @return [CARMA::Models::Submission] A persisted instance of Account
+      #
+      def self.from_claim(claim, metadata = {})
+        new(
+          data: claim.parsed_form,
+          metadata: metadata.merge(claim_id: claim.id)
+        )
+      end
+
+      def initialize(args = {})
+        @carma_case_id = args[:carma_case_id]
+        @submitted_at = args[:submitted_at]
+        @data = args[:data]
+
+        send('metadata=', args[:metadata] || {})
```

Comment:
Why not just
```suggestion
        @metadata = Metadata.new(args[:metadata] || {})
```
and remove the `metadata=` method?  Is this the `metadata=` method used anywhere else?

Reply:
I'm using `:send` so the logic to parse the setter value to the `Metadata` model, lives in one place. Although `:metadata=` is not being called anywhere, semantically I think it's appropriate for it to be a public method.

If you feel that I'm adding a method prematurely, I could remove `:metadata=` for now, and add it back in when it actually needs to be called by `/app`.

Reply:
I don't feel strongly.  But it is a public method, so you don't need to use `send`, right?
```suggestion
        self.metatdata = args[:metadata] || {}
```

Reply:
Yes, I can! Good call. I've updated it.